apollo-router/tests/integration/introspection.rs

```diff
@@ -178,16 +178,185 @@ async fn mixed() {
     "###);
 }
 
-async fn make_request(request: Request) -> apollo_router::graphql::Response {
-    apollo_router::TestHarness::builder()
-        .configuration_json(json!({
-            "supergraph": {
-                "introspection": true,
+const QUERY_DEPTH_2: &str = r#"{
+  __type(name: "Query") {
+    fields {
+      type {
+        fields {
+          type {
+            kind
+          }
+        }
+      }
+    }
+  }
+}"#;
+
+const QUERY_DEPTH_3: &str = r#"{
+  __type(name: "Query") {
+    fields {
+      type {
+        fields {
+          type {
+            fields {
+              name
+            }
+          }
+        }
+      }
+    }
+  }
+}"#;
+
+#[tokio::test]
+async fn just_under_max_depth() {
+    let request = Request::fake_builder()
+        .query(QUERY_DEPTH_2)
+        .build()
+        .unwrap();
+    let response = make_request(request).await;
+    insta::assert_json_snapshot!(response, @r###"
+    {
+      "data": {
+        "__type": {
+          "fields": [
+            {
+              "type": {
+                "fields": [
+                  {
+                    "type": {
+                      "kind": "NON_NULL"
+                    }
+                  },
+                  {
+                    "type": {
+                      "kind": "SCALAR"
+                    }
+                  },
+                  {
+                    "type": {
+                      "kind": "SCALAR"
+                    }
+                  },
+                  {
+                    "type": {
+                      "kind": "LIST"
+                    }
+                  }
+                ]
+              }
             },
-            "include_subgraph_errors": {
-                "all": true,
+            {
+              "type": {
+                "fields": null
+              }
+            }
+          ]
+        }
+      }
+    }
+    "###);
+}
+
+#[tokio::test]
+async fn just_over_max_depth() {
+    let request = Request::fake_builder()
+        .query(QUERY_DEPTH_3)
+        .build()
+        .unwrap();
+    let response = make_request(request).await;
+    insta::assert_json_snapshot!(response, @r###"
+    {
+      "errors": [
+        {
+          "message": "Maximum introspection depth exceeded",
+          "locations": [
+            {
+              "line": 7,
+              "column": 13
+            }
+          ]
+        }
+      ]
+    }
+    "###);
+}
+
+#[tokio::test]
+async fn just_over_max_depth_with_check_disabled() {
```

Comment:
is there a test for "over max depth with check enabled"? am i just not seeing it and scrolling past it?

Reply:
Yes it’s the one above, `async fn just_over_max_depth()`

Reply:
i think it's time to log off 🙃